### PR TITLE
meson: use template (part 4)

### PR DESCRIPTION
**Pull request by yuwata**
Split-out of #28042.

### CODE REVIEW

File: src/binfmt/meson.build [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+                'name' : 'systemd-binfmt',
+                'public' : true,
+                'conditions' : ['ENABLE_BINFMT'],
+                'sources' : files('binfmt.c'),
+        },
+]
+
+if conf.get('ENABLE_BINFMT') == 1
+        meson.add_install_script('sh', '-c',
+                         mkdir_p.format(binfmtdir))
```
Comment (medhefgo) [inline src/binfmt/meson.build:14]: There is `install_emptydir()` since 0.60.0...

**Review by medhefgo — APPROVED** [review]

File: src/binfmt/meson.build (hunk 1 above)
Comment (yuwata) [reply src/binfmt/meson.build:14]: Yeah. The last commit in #28042 replaces it.

File: meson.build [hunk 2]
```diff
@@ -2581,59 +2581,6 @@ meson.add_install_script(meson_make_symlink,
                          bindir / 'udevadm',
                          libexecdir / 'systemd-udevd')
 
-if conf.get('ENABLE_REMOTE') == 1 and conf.get('HAVE_LIBCURL') == 1
```
Comment (bluca) [inline meson.build:2682]: these were all optional and not built at all without the option before, is that still the case?

Comment (medhefgo) [reply meson.build:2682]: That's what the `conditions` parameter is for.

Comment (yuwata) [reply meson.build:2682]: Yes. Please see the `conditions` fields.

**Pull request merged**